src/Microsoft.AspNetCore.Server.Kestrel/Adapter/Internal/AdaptedPipeline.cs

```diff
@@ -42,21 +46,30 @@ public void Dispose()
 
             do
             {
-                var block = SocketInput.IncomingStart();
+                // TODO: We might want to read into tail space no matter how small it is
```

Comment:
I don't think the minimum size should be smaller than 2048. I would change the name to MinAllocBufferSize and remove the TODO.